analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/DebugLogger.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using SonarAnalyzer.CFG;
+using SonarAnalyzer.CFG.Roslyn;
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn;
+
+#pragma warning disable S106    // Standard outputs should not be used directly to log anything
+
+[ExcludeFromCodeCoverage]
+internal class DebugLogger
+{
+    private const string Separator = "----------";
+
+    private readonly bool isActive;
+
+#if DEBUG
```

Comment:
Do we need this? We consider this active only if a debugger is attached.

Reply:
It's defense in depth. I don't want to ship any line of code that depends on `Debugger`, because we often run inside VS instance from NuGet or SL and `Debugger` state could be funny. When debugging and modifying the code 😄